gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/PubSub.java

```diff
@@ -154,22 +162,29 @@ public static PullOption maxMessages(int maxMessages) {
 
   Future<AsyncPage<Subscription>> listSubscriptionsAsync(String topic, ListOption... options);
 
-  // Possible Ack options:
-  // 1) return a "special" iterator with "ack-iterated-messages"
-  // 2) provide a way to pull with callback(Message) - ask messages that were called successfully
-  //
-  // Also, consider auto-renewable for all messages that were pulled and not acked.
   List<ReceivedMessage> pull(String subscription, PullOption... options);
 
-  Future<List<Message>> pullAsync(String subscription, PullOption... options);
+  Future<List<ReceivedMessage>> pullAsync(String subscription, PullOption... options);
+
+  void pull(String subscription, PullCallback callback, PullOption... options);
```

Comment:
Just to make sure I understand. When a message is retrieved with:
```java
List<ReceivedMessage> pull(String subscription, PullOption... options);
```
Acking or nacking must be explicitly called by the user with `message.ack()` or `message.nack()`.
When a message is read with:
```java
void pull(String subscription, PullCallback callback, PullOption... options);
```
Acking and nacking is done automatically depending on `callback` return/exception.
Can you confirm or deny? :)